src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSourceProvider.cs

```diff
@@ -159,38 +165,54 @@ public IEnumerable<SuggestedActionSet> GetSuggestedActions(ISuggestedActionCateg
                     var optionService = workspace.Services.GetService<IOptionService>();
                     var supportSuggestion = workspace.Services.GetService<IDocumentSupportsSuggestionService>();
 
-                    IEnumerable<SuggestedActionSet> result = null;
-                    if (supportSuggestion.SupportsCodeFixes(document) && requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.CodeFix))
-                    {
-                        var suggestions = Task.Run(
-                            async () => await _owner._codeFixService.GetFixesAsync(document, range.Span.ToTextSpan(), cancellationToken).ConfigureAwait(false), cancellationToken).WaitAndGetResult(cancellationToken);
+                    var fixes = GetCodeFixes(
+                        supportSuggestion, requestedActionCategories,
+                        workspace, document, range, cancellationToken);
 
-                        result = OrganizeFixes(workspace, suggestions);
-                    }
+                    var refactorings = GetRefactorings(
+                        optionService, supportSuggestion, requestedActionCategories,
+                        workspace, document, snapshot, range, cancellationToken);
 
-                    if (supportSuggestion.SupportsRefactorings(document))
-                    {
-                        var refactoringResult = GetRefactorings(requestedActionCategories, document, snapshot, workspace, optionService, cancellationToken);
+                    return (fixes == null) ? refactorings :
+                                (refactorings == null) ? fixes : fixes.Concat(refactorings);
+                }
+            }
 
-                        result = result == null ? refactoringResult :
-                                    refactoringResult == null ? result :
-                                        result.Concat(refactoringResult);
+            private IEnumerable<SuggestedActionSet> GetCodeFixes(
+                IDocumentSupportsSuggestionService supportSuggestion,
+                ISuggestedActionCategorySet requestedActionCategories,
+                Workspace workspace,
+                Document document,
+                SnapshotSpan range,
+                CancellationToken cancellationToken)
+            {
+                IEnumerable<SuggestedActionSet> result = null;
+                if (_owner._codeFixService != null && supportSuggestion.SupportsCodeFixes(document))
+                {
+                    var includeSuppressionFixes = requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Any);
```

Comment:
Yes it is. If the light bulb is only for code fixes only (i.e. ```CODEFIX```), ```requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Any)``` will return false. However if it is explicitly asking for everything (```ANY```), then and only then should we include suppression fixes.